src/test/java/com/marklogic/kafka/connect/source/AbstractIntegrationSourceTest.java

```diff
@@ -92,25 +94,45 @@ void loadSingleAuthorRowIntoMarkLogicWithCustomTime(String uri, String id, Strin
     }
 
     void verifyQueryReturnsFifteenAuthors(List<SourceRecord> sourceRecords, String expectedValue) {
-        Assertions.assertEquals(15, sourceRecords.size());
+        verifyQueryReturnsFifteenAuthors(sourceRecords, expectedValue, "none");
+    }
+
+    void verifyQueryReturnsFifteenAuthors(List<SourceRecord> sourceRecords, String expectedValue, String keyStrategy) {
+        assertEquals(15, sourceRecords.size());
         assertTopicAndSingleValue(sourceRecords, expectedValue);
+        assertRecordKeys(sourceRecords, keyStrategy);
     }
 
     private void assertTopicAndSingleValue(List<SourceRecord> newSourceRecords, String expectedValue) {
         AtomicReference<Boolean> foundExpectedValue = new AtomicReference<>(false);
         newSourceRecords.forEach(sourceRecord -> {
-            Assertions.assertEquals(AUTHORS_TOPIC, sourceRecord.topic());
+            assertEquals(AUTHORS_TOPIC, sourceRecord.topic());
             assertTrue(sourceRecord.value() instanceof String, "Until we figure out how to return a JsonNode and make " +
                 "Confluent Platform happy, we expect the JsonNode to be toString'ed; type: " + sourceRecord.value().getClass());
             System.out.println(sourceRecord.value());
             if (expectedValue.equals(sourceRecord.value())) {
                 foundExpectedValue.set(true);
             }
         });
-        Assertions.assertTrue(foundExpectedValue.get(),
+        assertTrue(foundExpectedValue.get(),
             "List of SourceRecords does not contain a record with the expected value");
     }
 
+    private void assertRecordKeys(List<SourceRecord> newSourceRecords, String keyStrategy) {
+        if(keyStrategy.equalsIgnoreCase("uuid")) {
+            newSourceRecords.forEach(sourceRecord -> assertEquals(UUID.fromString(sourceRecord.key().toString()).toString(), sourceRecord.key().toString()));
+        } else if(keyStrategy.equalsIgnoreCase("timestamp")) {
+            AtomicLong rowNumber = new AtomicLong(1);
```

Comment:
This becomes a more interesting assertion if we use the ML server timestamp, as we could verify that each key begins with the same timestamp.

Reply:
No action needed for now.